crates/subspace-networking/src/request_handlers/root_block.rs

```diff
@@ -0,0 +1,33 @@
+//! Helper for incoming root block requests.
+//!
+//! Handle (i.e. answer) incoming root blocks requests from a remote peer received via
+//! `RequestResponsesBehaviour` with generic [`GenericRequestHandler`].
+
+use crate::request_handlers::generic_request_handler::{GenericRequest, GenericRequestHandler};
+use parity_scale_codec::{Decode, Encode};
+use subspace_core_primitives::{RootBlock, SegmentIndex};
+
+/// Root block by segment indexes protocol request.
+#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode)]
+pub struct RootBlockRequest {
+    /// Request key - piece index hash
+    pub segment_indexes: Vec<SegmentIndex>,
+}
+
+impl GenericRequest for RootBlockRequest {
+    const PROTOCOL_NAME: &'static str = "/subspace/root-blocks-by-seg-index/0.1.0";
```

Comment:
Can we change this to segment header too since this is a long-term identifier, please?